crates/cast/src/helpers/scarb_utils.rs

```diff
@@ -12,31 +11,30 @@ pub struct CastConfig {
     pub rpc_url: String,
     pub network: String,
     pub account: String,
+    pub accounts_file: Option<Utf8PathBuf>,
```

Comment:
This should just be `UtfPathBuf`, and if the value is not found in the config, we must take care of it in main (as it is with the rest of fields from this struct)

Reply:
Any arguments on why this should be just `Utf8PathBuf`?
I thought it was an optional field, that could either be present in the profile or not.

Reply:
We do not have to set parameters as Optional, because CastConfig derives `Default` trait (see line 10 in this file), which means if something is not set, it gets the type default.

Now CastConfig is being constructed in main (`parse_scarb_config`), and in there it represents the final config state - and in this context this path is required (as well as other parameters - you can't run cast without rpc url). So this is the place where we currently do all the shenanigans regarding config - so, we look if there is something (anythong) configured in Scarb.toml, we check if something is being set using CLI, pick precedence etc.

We should rather stick to one approach - if you feel things should change, because they could be better - do it, and let's see how it is. Otherwise, let's stick to one approach :+1:

Reply:
I was thinking already forward, as if to stick with this approach, other types would likely be of Option type as well.
To me it seemed that only the first 3 (rpc_url, network, account) would be required.

So is CastConfig supposed to be the actual config on which the program executes stuff, or just the config that is contained in Scarb.toml?
The current approach would suggest the second, as later, the config variable is not updated with the arguments passed through CLI, which have precedence over arguments from Scarb.toml.

Reply:
I see, in that case this is ok. Out of those 3, probably network shouldn't be required as well, since we can retrieve this from the provider.

Sure, the second approach is kind of what we have now. It could also make sense if we moved towards the first option though, I suppose. If you have a strong opinion that you would like to share, please do so. I'm honestly fine with both, but haven't given it much thought (and maybe I should :D ).

Reply:
I feel like it would be easier to create a single config that the current execution runs on and then just pass it around (so the first approach - to me it would make sense too).

Reply:
Sure, I like this. Do you want to handle this in this PR, or create another issue? both ways are fine for me

Reply:
I think I'll try to fiddle with it here, and later think about splitting it, if it were to turn out a big, confusing for this pr change